core/state/trie_db.go

```diff
@@ -0,0 +1,952 @@
+package state
```

Comment:
can we move it somewhere to `zk` package?

Reply:
Majority of the code in this file was copied from [core/state/database.go](https://github.com/0xPolygonZero/erigon/blob/feat/zero/core/state/database.go). I think it makes sense to keep this in state module because it will be used not only by SMT, but also MPT as well when we merge two repos. I put the code in a new file (trie_db.go), so it will bring less conflict when merging in the future. I can move it to `zk` package if you still think it will be better to do so.

Reply:
sounds good